### PR TITLE
expression: rewrite builtin function: AES_ENCRYPT, AES_DECRYPT

### PR DESCRIPTION
**Pull request by breeswish**
~~Note: this PR does **NOT** fix https://github.com/pingcap/tidb/issues/4084~~
Update: this PR also fixes https://github.com/pingcap/tidb/issues/4084 now.

@XuHuaiyu @zz-jason PTAL

### CODE REVIEW

**Comment by breeswish** [comment]
https://github.com/pingcap/tidb/issues/4080

File: expression/builtin_encryption.go [hunk 1]
```diff
@@ -83,93 +83,82 @@ type aesDecryptFunctionClass struct {
 }
 
 func (c *aesDecryptFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	err := errors.Trace(c.verifyArgs(args))
-	bt := &builtinAesDecryptSig{newBaseBuiltinFunc(args, ctx)}
-	bt.deterministic = true
-	return bt.setSelf(bt), errors.Trace(err)
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpString)
```
Comment (zz-jason) [inline expression/builtin_encryption.go]: `verifyArgs(args)` firstly

File: expression/builtin_encryption.go [hunk 2]
```diff
@@ -83,93 +83,82 @@ type aesDecryptFunctionClass struct {
...
 
 type aesEncryptFunctionClass struct {
 	baseFunctionClass
 }
 
 func (c *aesEncryptFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	err := errors.Trace(c.verifyArgs(args))
-	bt := &builtinAesEncryptSig{newBaseBuiltinFunc(args, ctx)}
-	return bt.setSelf(bt), errors.Trace(err)
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpString)
```
Comment (zz-jason) [inline expression/builtin_encryption.go]: ditto

**Comment by breeswish** [comment]
@zz-jason @XuHuaiyu PTAL

**Comment by zz-jason** [comment]
@breeswish plz resolve conflicts

**Pull request merged**